activerecord/test/cases/tasks/postgresql_rake_test.rb

```diff
@@ -139,45 +181,70 @@ def drop_database(*); end
           "adapter"  => "postgresql",
           "database" => "my-app-db"
         }
-
-        ActiveRecord::Base.stubs(:connection).returns(@connection)
-        ActiveRecord::Base.stubs(:establish_connection)
       end
 
       def test_clears_active_connections
-        ActiveRecord::Base.expects(:clear_active_connections!)
+        with_stubbed_connection_clear_active_connections do
+          ActiveRecord::Base.stub(:establish_connection, true) do
+            ActiveRecord::Base.expects(:clear_active_connections!)
 
-        ActiveRecord::Tasks::DatabaseTasks.purge @configuration
+            ActiveRecord::Tasks::DatabaseTasks.purge @configuration
+          end
+        end
       end
 
       def test_establishes_connection_to_postgresql_database
-        ActiveRecord::Base.expects(:establish_connection).with(
-          "adapter"            => "postgresql",
-          "database"           => "postgres",
-          "schema_search_path" => "public"
-        )
-
-        ActiveRecord::Tasks::DatabaseTasks.purge @configuration
+        ActiveRecord::Base.stubs(:establish_connection).returns(true)
```

Comment:
Can we get rid of `ActiveRecord::Base.stubs(:establish_connection).returns(true)` here?

Reply:
We can get rid of the return value. Still need ActiveRecord::Base.respond_to?(:establish_connection) == true. Fixing.